### ISSUE TITLE
`findAndObserve(id)` throws an uncatchable error when id does not exist

### ISSUE THREAD

**Issue by SaltedBlowfish** [Nozbe/WatermelonDB#1257]
### Current Behavior

Error: `Diagnostic error: Record posts#abcde not found`

```
React.useEffect(() => {
  const thisIdDoesNotExist = 'abcde';

  try {
    const subscription = database
      .get<Post>('posts')
      .findAndObserve(thisIdDoesNotExist)
      .subscribe(d => {
        // Adding try...catch block here also does not catch
      });

    return () => subscription.unsubscribe();
  } catch (e: any) {
    console.warn(e); // <-- Never gets called
  }
}, []);
```

### Expected Behavior

The try...catch block catches the database error about the record not existing.

### Workaround

```
React.useEffect(() => {
  const thisIdDoesNotExist = 'abcde';

  try {
    const subscription = database
      .get<Post>('posts')
      .query(Q.where('id', thisIdDoesNotExist), Q.take(1))
      .observe()
      .subscribe(d => {
        setPost(d[0]); // <-- Just take 0th record since query returns at most 1
      });

    return () => subscription.unsubscribe();
  } catch (e: any) {
    console.warn(e); // <-- Never gets called
  }
}, []);
```

**Comment by KrisLau** [comment]
I just swapped my find query for a regular `.query(Q.where())` so I can check if there is no matching records. Not the ideal method but a workaround for now

**Comment by SaltedBlowfish** [comment]
> I just swapped my find query for a regular `.query(Q.where())` so I can check if there is no matching records. Not the ideal method but a workaround for now

So you did what I documented in the workaround above or did you do something different?

**Comment by avinashlng1080** [comment]
@SaltedBlowfish  your workaround looks great.  It would seem that if `findAndObserve` encounters an error, then it would stop observing as well.  Have you encountered this situation?

**Comment by avinashlng1080** [comment]
Another solution - try to shield the main observable chain from the catch.  i.e. add the catch to an inner switchMap.  That way, observation on the main chain will keep alive even if the inner chain dies.  See this [link](https://iamturns.com/continue-rxjs-streams-when-errors-occur/) for more info.

**Comment by pronevich** [comment]
@avinashlng1080 this doesn't work, the error happens in `findAndObserve` and is thrown before `switchMap`. In linked article - slightly another case.

**Comment by moriax** [comment]
I created a small patch, which added a new paramter to findAndObserve which gives me the capability to not throw an error in specific use cases.
![image](https://user-images.githubusercontent.com/12937605/161432493-a0c3ceba-985b-4423-9f12-90fd9914b770.png)

You think something similliar could be added to this package ?

**Comment by nhanders** [comment]
I'm not fantastic with RxJs, but I found I could handle these diagnostic errors within the `subscribe` method:

```
const subscription = database
  .get<Post>('posts')
  .findAndObserve(thisIdDoesNotExist)
  .subscribe({
    next: (d) => handleUpdate(d),
    error: (error) => handleError(error),
  });
```

**Comment by barbalex** [comment]
> It would seem that if `findAndObserve` encounters an error, then it would stop observing as well. Have you encountered this situation?

@avinashlng1080 
Yes, that is exactly what we ran into. In our app data is synced from the server to the client. If a client opens a link that leads to a specific id, not all data has been synced yet so this happens. Because observing stops the data does not appear even after finishing syncing.

**Comment by 3210jr** [comment]
Just ran into this as well ... in production. Here is how i handle it in a withObserverables hook:

```
const enhance = withObservables(["item"], ({ item }: { item: AppointmentModel }) => ({
  item,
  client: item.client?.observe().pipe(catchError(() => of$(null))),
}))
```

The client code threw an error before, when the client no longer existed for one reason or another. This seems to work for me, lmk if this works
